clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -86,6 +85,25 @@ public AbortedTransaction(long pid, long firstOffset) {
         }
 
         @Override
+        public boolean equals(Object o) {
+            if (this == o)
+                return true;
+            if (o == null || getClass() != o.getClass())
+                return false;
+
+            AbortedTransaction that = (AbortedTransaction) o;
+
+            return pid == that.pid && firstOffset == that.firstOffset;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = (int) (pid ^ (pid >>> 32));
```

Comment:
I am not strong which one is more important but I am also not convinced that we are concerned with performance here, particularly since this hashCode and equals seem to be only used in test. That is why I pinged two more person for quick comment. After all this is not an urgent bug fix and we have time to wait for comment.

I get that we use this old fashioned way somewhere in Kafka. But it doesn't mean it is the best way going forward. In general we want to reuse exisiting library call and don't really care about performance of code used only but test. Just my 2 cents.

I get that we need this only in FetchResponse because we only have test that covers this for FetchResponse. This implies that we still don't have test that covers similar thing and other request like ProduceRequest. Is this something that we need to fix, maybe in a future patch? After all it is inconsistent that we only override this for FetchResponse. We can do it in another patch. What do you think?